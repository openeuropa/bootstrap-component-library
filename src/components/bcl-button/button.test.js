import { merge, renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-data-button/data";
import toggleDemoData from "@openeuropa/bcl-data-button/toggleData";
import popoverDemoData from "@openeuropa/bcl-data-button/popoverData";
import tooltipDemoData from "@openeuropa/bcl-data-button/tooltipData";
import spinnerDemoData from "@openeuropa/bcl-data-button/spinnerData";
import { getVariants } from "@openeuropa/bcl-story-utils";
import drupalAttribute from "drupal-attribute";

const template = "@oe-bcl/bcl-button/button.html.twig";
const render = (params, reset) => renderTwigFileAsNode(template, params, reset);
const variants = getVariants(false, ["link"]);
const sizes = ["lg", "md", "sm"];

describe("OE - Button", () => {
  variants.forEach((variant) => {
    test(`${variant} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render(
          {
            ...demoData,
            variant: variant,
          },
          true
        )
      ).resolves.toMatchSnapshot();
    });
  });

  sizes.forEach((size) => {
    test(`${size} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({ ...demoData, size: size }, true)
      ).resolves.toMatchSnapshot();
    });
  });

  test("with toggle renders correctly", () => {
    expect.assertions(1);

    return expect(render(toggleDemoData)).resolves.toMatchSnapshot();
  });

  test("with spinner renders correctly", () => {
    expect.assertions(1);

    return expect(render(spinnerDemoData)).resolves.toMatchSnapshot();
  });

  test("with icon renders correctly", () => {
    expect.assertions(1);

    return expect(
      render(
        {
          ...demoData,
          icon: {
            name: "mouse",
            transformation: "rotate-90",
            path: "bootstrap-icons.svg",
            attributes: new drupalAttribute(),
          },
        },
        true
      )
    ).resolves.toMatchSnapshot();
  });

  test("with popover renders correctly", () => {
    expect.assertions(1);

    return expect(render(popoverDemoData)).resolves.toMatchSnapshot();
  });

  test("with tooltip renders correctly", () => {
    expect.assertions(1);

    return expect(render(tooltipDemoData)).resolves.toMatchSnapshot();
  });
});

describe("OE - Button Outline", () => {
  variants.forEach((variant) => {
    test(`${variant} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render(
          {
            ...demoData,
            variant: variant,
            outline: true,
          },
          true
        )
      ).resolves.toMatchSnapshot();
    });
  });
});
