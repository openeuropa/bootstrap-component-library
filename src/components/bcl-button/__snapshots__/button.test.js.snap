// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Button Outline danger renders correctly 1`] = `
<jest>
  <button
    class="btn btn-outline-danger btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button Outline dark renders correctly 1`] = `
<jest>
  <button
    class="btn btn-outline-dark btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button Outline info renders correctly 1`] = `
<jest>
  <button
    class="btn btn-outline-info btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button Outline light renders correctly 1`] = `
<jest>
  <button
    class="btn btn-outline-light btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button Outline link renders correctly 1`] = `
<jest>
  <button
    class="btn btn-outline-link btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button Outline primary renders correctly 1`] = `
<jest>
  <button
    class="btn btn-outline-primary btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button Outline secondary renders correctly 1`] = `
<jest>
  <button
    class="btn btn-outline-secondary btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button Outline success renders correctly 1`] = `
<jest>
  <button
    class="btn btn-outline-success btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button Outline warning renders correctly 1`] = `
<jest>
  <button
    class="btn btn-outline-warning btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button danger renders correctly 1`] = `
<jest>
  <button
    class="btn btn-danger btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button dark renders correctly 1`] = `
<jest>
  <button
    class="btn btn-dark btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button info renders correctly 1`] = `
<jest>
  <button
    class="btn btn-info btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button lg renders correctly 1`] = `
<jest>
  <button
    class="btn btn-primary btn-lg"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button light renders correctly 1`] = `
<jest>
  <button
    class="btn btn-light btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button link renders correctly 1`] = `
<jest>
  <button
    class="btn btn-link btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button md renders correctly 1`] = `
<jest>
  <button
    class="btn btn-primary btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button primary renders correctly 1`] = `
<jest>
  <button
    class="btn btn-primary btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button secondary renders correctly 1`] = `
<jest>
  <button
    class="btn btn-secondary btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button sm renders correctly 1`] = `
<jest>
  <button
    class="btn btn-primary btn-sm"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button success renders correctly 1`] = `
<jest>
  <button
    class="btn btn-success btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button warning renders correctly 1`] = `
<jest>
  <button
    class="btn btn-warning btn-md"
    type="button"
  >
    Label
  </button>
</jest>
`;

exports[`OE - Button with icon renders correctly 1`] = `
<jest>
  <button
    class="btn btn-primary btn-md"
    type="button"
  >
    Label
    <svg
      class="ms-2-5 bi icon--fluid icon--rotate-90"
    >
      <use
        xlink:href="bootstrap-icons.svg#mouse"
      />
    </svg>
  </button>
</jest>
`;

exports[`OE - Button with popover renders correctly 1`] = `
<jest>
  <button
    autocomplete="off"
    class="btn btn-primary btn-md"
    data-bs-content="This is a popover"
    data-bs-placement="top"
    data-bs-toggle="popover"
    data-bs-trigger="focus"
    title="This is a title of popover"
    type="button"
  >
    Button with popover
  </button>
</jest>
`;

exports[`OE - Button with spinner renders correctly 1`] = `
<jest>
  <button
    class="btn btn-primary btn-md"
    type="button"
  >
    <div
      class="me-1 spinner-border spinner-border-sm text-info"
      role="status"
    >
      <span
        class="visually-hidden"
      >
        Loading...
      </span>
    </div>
    Label
  </button>
</jest>
`;

exports[`OE - Button with toggle renders correctly 1`] = `
<jest>
  <button
    aria-controls="collapseExample"
    aria-expanded="false"
    autocomplete="off"
    class="btn btn-primary btn-md"
    data-bs-target="#collapseExample"
    data-bs-toggle="collapse"
    type="button"
  >
    Button with collapse
  </button>
</jest>
`;

exports[`OE - Button with tooltip renders correctly 1`] = `
<jest>
  <button
    autocomplete="off"
    class="btn btn-primary btn-md"
    data-bs-toggle="tooltip"
    title="This is a tooltip"
    type="button"
  >
    Button with tooltip
  </button>
</jest>
`;
